src/gui/SampleWaveform.cpp

```diff
@@ -28,41 +28,50 @@ namespace lmms::gui {
 
 void SampleWaveform::visualize(Parameters parameters, QPainter& painter, const QRect& rect)
 {
-	const auto x = rect.x();
-	const auto height = rect.height();
-	const auto width = rect.width();
-	const auto centerY = rect.center().y();
+	const int x = rect.x();
+	const int height = rect.height();
+	const int width = rect.width();
+	const int centerY = rect.center().y();
 
-	const auto halfHeight = height / 2;
+	const int halfHeight = height / 2;
```

Comment:
I believe these should stay auto unless the type returned are not int and we specifically want int.

Reply:
Since these types are going to be used within calculations later in the function, I think that we should specify the types here. We should really specify the type for our primitive variables since we do know exactly what type we want and its helps to avoid bugs in which the wrong type was used.